From 73fe47532572dfb4d75426796776e0cafc6e1a61 Mon Sep 17 00:00:00 2001
From: Adrien Plazas <kekun.plazas@laposte.net>
Date: Wed, 1 Sep 2021 11:47:28 +0200
Subject: [PATCH] metered-data-dialog: Make it a GsInfoWindow and a
 HdyStatusPage

This ensures it will fit on phones, makes the code simpler, and makes it
look nicer.
---
 src/gs-metered-data-dialog.c  |  6 +--
 src/gs-metered-data-dialog.h  |  5 ++-
 src/gs-metered-data-dialog.ui | 75 +++++++++--------------------------
 src/gs-shell.c                |  8 +---
 4 files changed, 26 insertions(+), 68 deletions(-)

diff --git a/src/gs-metered-data-dialog.c b/src/gs-metered-data-dialog.c
index 9cd66987c..7e8a35baf 100644
--- a/src/gs-metered-data-dialog.c
+++ b/src/gs-metered-data-dialog.c
@@ -16,12 +16,12 @@
 
 struct _GsMeteredDataDialog
 {
-	GtkDialog	 parent_instance;
+	GsInfoWindow	 parent_instance;
 
 	GtkWidget	*button_network_settings;
 };
 
-G_DEFINE_TYPE (GsMeteredDataDialog, gs_metered_data_dialog, GTK_TYPE_DIALOG)
+G_DEFINE_TYPE (GsMeteredDataDialog, gs_metered_data_dialog, GS_TYPE_INFO_WINDOW)
 
 static void
 button_network_settings_clicked_cb (GtkButton *button,
@@ -60,9 +60,7 @@ gs_metered_data_dialog_new (GtkWindow *parent)
 	GsMeteredDataDialog *dialog;
 
 	dialog = g_object_new (GS_TYPE_METERED_DATA_DIALOG,
-			       "use-header-bar", TRUE,
 			       "transient-for", parent,
-			       "modal", TRUE,
 			       NULL);
 
 	return GTK_WIDGET (dialog);
diff --git a/src/gs-metered-data-dialog.h b/src/gs-metered-data-dialog.h
index 8219e9dce..17e177747 100644
--- a/src/gs-metered-data-dialog.h
+++ b/src/gs-metered-data-dialog.h
@@ -11,12 +11,15 @@
 #include <glib.h>
 #include <glib-object.h>
 #include <gtk/gtk.h>
+#include <handy.h>
+
+#include "gs-info-window.h"
 
 G_BEGIN_DECLS
 
 #define GS_TYPE_METERED_DATA_DIALOG (gs_metered_data_dialog_get_type ())
 
-G_DECLARE_FINAL_TYPE (GsMeteredDataDialog, gs_metered_data_dialog, GS, METERED_DATA_DIALOG, GtkDialog)
+G_DECLARE_FINAL_TYPE (GsMeteredDataDialog, gs_metered_data_dialog, GS, METERED_DATA_DIALOG, GsInfoWindow)
 
 GtkWidget	*gs_metered_data_dialog_new	(GtkWindow	*parent);
 
diff --git a/src/gs-metered-data-dialog.ui b/src/gs-metered-data-dialog.ui
index 085cb78cf..043c178f9 100644
--- a/src/gs-metered-data-dialog.ui
+++ b/src/gs-metered-data-dialog.ui
@@ -1,69 +1,30 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <interface>
   <requires lib="gtk+" version="3.10"/>
-  <template class="GsMeteredDataDialog" parent="GtkDialog">
+  <template class="GsMeteredDataDialog" parent="GsInfoWindow">
     <property name="title" translatable="yes">Automatic Updates Paused</property>
-    <property name="modal">True</property>
-    <property name="destroy-with-parent">True</property>
-    <property name="resizable">False</property>
-    <property name="type-hint">dialog</property>
-    <property name="skip-taskbar-hint">True</property>
-    <property name="use-header-bar">1</property>
-    <child internal-child="headerbar">
-      <object class="GtkHeaderBar">
-        <child type="title">
-          <object class="GtkLabel">
-            <property name="visible">True</property>
-            <property name="label" translatable="yes">Automatic Updates Paused</property>
-            <property name="selectable">False</property>
-            <style>
-              <class name="title"/>
-            </style>
-          </object>
-        </child>
-      </object>
-    </child>
-    <child internal-child="vbox">
-          <object class="GtkBox">
-            <property name="visible">True</property>
-            <property name="orientation">vertical</property>
-            <property name="spacing">12</property>
-            <property name="margin">12</property>
-            <child>
-              <object class="GtkLabel">
-                <property name="visible">True</property>
-                <property name="justify">left</property>
-                <property name="wrap">True</property>
-                <property name="wrap-mode">word-char</property>
-                <property name="label" translatable="yes">The current network is metered. Metered connections have data limits or charges associated with them. To save data, automatic updates have therefore been paused.
+    <child>
+      <object class="HdyStatusPage">
+        <property name="visible">True</property>
+        <property name="title" bind-source="GsMeteredDataDialog" bind-property="title" bind-flags="sync-create"/>
+        <property name="description" translatable="yes">The current network is metered. Metered connections have data limits or charges associated with them. To save data, automatic updates have therefore been paused.
 
 Automatic updates will be resumed when an un­metered network becomes available. Until then, it is still possible to manually install updates.
 
 Alternatively, if the current network has been in­correctly identified as being metered, this setting can be changed.</property>
-                <property name="max-width-chars">50</property>
-                <property name="halign">center</property>
-              </object>
-            </child>
-            <child>
-              <object class="GtkBox">
-                <property name="visible">True</property>
-                <child type="center">
-                  <object class="GtkButton" id="button_network_settings">
-                    <property name="visible">True</property>
-                    <property name="label" translatable="yes">Open Network _Settings</property>
-                    <property name="can-focus">True</property>
-                    <property name="receives-default">True</property>
-                    <property name="use-underline">True</property>
-                    <signal name="clicked" handler="button_network_settings_clicked_cb"/>
-                  </object>
-                </child>
-              </object>
-              <packing>
-                <property name="fill">False</property>
-                <property name="expand">False</property>
-              </packing>
-            </child>
+        <property name="icon-name">network-cellular-signal-excellent-symbolic</property>
+        <child>
+          <object class="GtkButton" id="button_network_settings">
+            <property name="visible">True</property>
+            <property name="label" translatable="yes">Open Network _Settings</property>
+            <property name="can-focus">True</property>
+            <property name="halign">center</property>
+            <property name="receives-default">True</property>
+            <property name="use-underline">True</property>
+            <signal name="clicked" handler="button_network_settings_clicked_cb"/>
           </object>
+        </child>
+      </object>
     </child>
   </template>
 </interface>
diff --git a/src/gs-shell.c b/src/gs-shell.c
index d81657d38..16b4626e1 100644
--- a/src/gs-shell.c
+++ b/src/gs-shell.c
@@ -263,14 +263,10 @@ gs_shell_metered_updates_bar_response_cb (GtkInfoBar *info_bar,
 					  gpointer    user_data)
 {
 	GsShell *shell = GS_SHELL (user_data);
-	GtkDialog *dialog;
+	GtkWidget *dialog;
 
-	dialog = GTK_DIALOG (gs_metered_data_dialog_new (GTK_WINDOW (shell)));
+	dialog = gs_metered_data_dialog_new (GTK_WINDOW (shell));
 	gs_shell_modal_dialog_present (shell, GTK_WINDOW (dialog));
-
-	/* just destroy */
-	g_signal_connect_swapped (dialog, "response",
-				  G_CALLBACK (gtk_widget_destroy), dialog);
 }
 
 static void
-- 
2.30.2

